### DISCUSSION TITLE
Door controller (d1 mini)

### DISCUSSION THREAD

**Discussion by antonpiatek** [arendst/Tasmota#17480] [Q&A]
I'm looking to build a door controller based on a d1 mini running tasmota (for a chicken coop). The blinds controller seems like a good starting point as one gpio pin for each of up/down will control the door via an external motor controller. One thing I can't see in the docs, is any form of sensor to know if the door is fully opened or closed? It seems very valuable to me to know that that door is actually close (and open), but the docs don't seem to mention any sensor inputs to trigger this. Am I missing something in the docs? How does it know when powered on what state is it in?

The hardware is literally just a geared motor with a string to pull the door, I'm not sure how well this will work without some absolute position information - is there a better hardware approach than simply winding up a string for the door? It has a little weight as is wood, and has to move a fair distance so a servo seems out.

Thanks for any help/ideas - I had this running previously on a raspberry pi and node-red, but tasmota has such nice status info and integrations I wanted to rebuild as a tasmota module (also dedicated hardware is a bit more reliable, and tasmota is a solid base)

**Comment by sfromis** [comment]
Blinds are usually calibrated for precise positioning, with known limits to match the window, and to get to a predetermined spot when commanded. A door controller may be more like a garage controller, which indeed typically includes a reed switch to say stop. Much simpler than the complexities of handling precise blinds behavior.

**Reply by antonpiatek** [reply]
Yes, exactly. I have a top and bottom reed switch already.
Is there a tasmota mode which suits this operation?
